app/src/main/java/woowacourse/payments/domain/model/CardNumber.kt

```diff
@@ -0,0 +1,16 @@
+package woowacourse.payments.domain.model
+
+import woowacourse.payments.domain.exception.InvalidCardException
+
+@JvmInline
+value class CardNumber private constructor(
+    val value: String,
+) {
+    companion object {
+        private const val CARD_NUMBER_LENGTH = 16
+
+        fun from(raw: String): CardNumber? = if (raw.length == CARD_NUMBER_LENGTH) CardNumber(raw) else null
+
+        fun require(raw: String): CardNumber = from(raw) ?: throw InvalidCardException.InvalidCardNumber()
+    }
+}
```

Comment:
```suggestion
value class CardNumber(
    val value: String,
) {
    init {
        if (value.length != CARD_NUMBER_LENGTH) {
            throw InvalidCardException.InvalidCardNumber()
        }
    }

    companion object {
        private const val CARD_NUMBER_LENGTH = 16
    }
}

```

이 코드와 비교해서 비비가 작성해주신 코드는 어떠한 장점이 있나요?🧐

Reply:
제가 작성한 코드에서는 외부에서는 `CardNumber("~~")`와 같이 직접 생성이 불가하고, 반드시 `from` 혹은 `require`로만 생성을 하도록 했습니다. 이렇게 무조건 팩토리를 거치게 하므로 유효성 로직을 한 곳에서 통제할 수 있다는 장점이 있습니다!

저는 특히 `CardNumber("~~")`를 아예 막는 점이 좋다고 생각했는데 제리는 init에서 유효성 검증을 하는 것이 더 좋다고 생각하시나요?? 제리의 의견이 궁금합니다!